packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts

```diff
@@ -427,6 +432,36 @@ export class StripeWebhookHandler extends StripeHandler {
     await this.updateCustomer(uid, email);
   }
 
+  /**
+   * Validate product metadata on update
+   */
+  async handleProductUpdate(request: AuthRequest, event: Stripe.Event) {
+    const product = event.data.object as Stripe.Source;
+
+    const plans = this.stripeHelper.fetchPlansByProductId(product.id);
```

Comment:
I don't think `product.id` exists if this executes during a `plan.updated` event.  Also, we don't need to validate all the plans when only one plan has changed.  We only need to do that for `product.updated`.